sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -384,7 +407,7 @@ case class ExpressionEncoder[T](
 
   override def toString: String = s"class[$schemaString]"
 
-  override def makeCopy: ExpressionEncoder[T] = copy()
+  override def makeCopy: ExpressionEncoder[T] = this
```

Comment:
do we still need this API?

Reply:
It was added by https://issues.apache.org/jira/browse/SPARK-28456, in 3.0

Reply:
No. I can try to remove it, but we might break a 'public' API.

Reply:
Oh ok, if we port to 3.0 then it is safe to remove.